### DISCUSSION TITLE
React Rourter

### DISCUSSION THREAD

**Discussion by Parth1508** [vercel/next.js#36831] [Help]
Hello,
if the user opens a page that needs authentication I push him to the login page, but I need to push him back to this page
how I can achieve that

**Comment by icyJoseph** [comment]
You can call `router.back()`, or alternatively....

```tsx
import Router, { useRouter } from "next/router";
import { useEffect } from "react";

const Home: NextPage = () => {
  const router = useRouter();
  const { asPath } = router;

  const user = null;

  useEffect(() => {
    if (user) return;

    // let's say this is how you know they need to be authenticated (if you do it client side)
    Router.push({ pathname: "/login", query: {returnPath: asPath} }, "/login");
   // if this page/component doesn't unmount, this might end up in an infinite loop
  }, [asPath, user]);

  return <div></div>;
};
```

Then from `/login`, when they successfully logged in:

```ts
const onSuccess = () => {
    router.replace({pathname: router.query.returnPath || '/'}); // fallback on landing page
}
```

Something like that... if you are doing the authentication redirect on the server (GSSP) or other, you could just include the query with `returnPath` in the URL.

Be mindful that redirecting like this, could be used against your users, so always check that `returnPath` at least, `startWith("/")` so that it is a route, or make sure if one of the pages you know.
